### ISSUE TITLE
Garbage collection goes through all namespaces?

### ISSUE THREAD

**Issue by gytisgreitai** [ksonnet/ksonnet#244]
I have a `test` environment defined:

```
{
  "server": "https://xx.xx.xx.xx",
  "namespace": "test"
}
```

Though if I run :
`ks -v apply --gc-tag "ksnt" test`

It seems that it is walking though all namespaces:

```
< .. snip ..>
DEBUG Considering deployments default.es-stats-exporter (apps/v1beta1) for gc
DEBUG Considering deployments default.external-dns (apps/v1beta1) for gc
< .. snip ..>
```

**Comment by bryanl** [comment]
In the fix for #200, we're going to change how we are applying changes (which includes garbage collection). Stay tuned for a fix.

**Comment by bryanl** [comment]
Update: we are refining how ksonnet wordpks and have decided that an environment is only one namespace. This is a bug and will be addressed.
